### PR TITLE
[Wip] lapi integration

### PR DESCRIPTION
**Pull request by buixor**
- Makefile(s) build api as well
 - Fix debug levels
 - Fix models.Alert build

### CODE REVIEW

File: cmd/crowdsec/main.go [hunk 1]
```diff
@@ -357,18 +357,9 @@ func main() {
...
-	// 	if len(parsers.nodes) == 0 {
-	// 		log.Fatalf("no parser(s) loaded, abort.")
-	// 	}
-
-	// 	if len(holders) == 0 {
-	// 		log.Fatalf("no bucket(s) loaded, abort.")
-	// 	}
-	// }
+	if cConfig.API == nil || cConfig.API.Client == nil || cConfig.API.Client.Credentials == nil {
+		log.Fatalf("Missing Client API credentials, abort.")
```
Comment (erenJag) [inline cmd/crowdsec/main.go:361]: We need to be more specific on which config is missing (local or online API credentials)

File: pkg/leakybucket/overflows.go [hunk 2]
```diff
@@ -126,73 +190,61 @@ func NewAlert(leaky *Leaky, queue *Queue) types.RuntimeAlert {
...
+	if err != nil {
+		return runtimeAlert, errors.Wrap(err, "unable to collect sources from bucket")
+	}
+	runtimeAlert.Sources = sources
+	//Include source info in format string
+	sourceStr := ""
+	if len(sources) > 1 {
+		sourceStr = fmt.Sprintf("%d Sources on scope.", len(sources))
+	} else if len(sources) == 1 {
+		for k, _ := range sources {
```
Comment (registergoofy) [inline pkg/leakybucket/overflows.go:214]: Use `for k := range sources` here

File: pkg/parser/enrich.go [hunk 3]
```diff
@@ -44,7 +44,6 @@ func Loadplugin(path string) ([]EnricherCtx, error) {
 	if err != nil {
 		log.Warningf("load (fake) plugin load : %v", err)
 		c.initiated = false
-		return nil, err
```
Comment (registergoofy) [inline pkg/parser/enrich.go:47]: Why ? I think we should fatal if this fails ?

**Review by registergoofy — CHANGES_REQUESTED** [review]
minor changes.

File: Makefile [hunk 4]
```diff
@@ -6,6 +6,7 @@ DATA_PREFIX = $(PREFIX)"/var/run/crowdsec/"
 PID_DIR = $(PREFIX)"/var/run/"
 CROWDSEC_FOLDER = "./cmd/crowdsec"
 CSCLI_FOLDER = "./cmd/crowdsec-cli/"
+API_FOLDER= "./cmd/api/"
```
[the remark this answers is not in the document]
Comment (buixor) [reply Makefile:9]: yes it is for testing purpose, it's going to get removed once the mains are merge

File: cmd/crowdsec/output.go [hunk 5]
```diff
@@ -138,9 +137,10 @@ LOOP:
 				return fmt.Errorf("postoverflow failed : %s", err)
 			}
 
-			if *event.Overflow.Alert.Scenario == "" && event.Overflow.Mapkey != "" {
+			if event.Overflow.Alert == nil && event.Overflow.Mapkey != "" {
 				buckets.Bucket_map.Delete(event.Overflow.Mapkey)
-			} else {
+			} else if event.Overflow.Alert != nil {
+				log.Printf("%s", *event.Overflow.Alert.Message)
```
[the remark this answers is not in the document]
Comment (buixor) [reply cmd/crowdsec/output.go:143]: no, it really displays (Printf == Infof) :

```
INFO[28-09-2020 12:09:00] Ip 127.0.0.1 performed 'crowdsecurity/http-bad-user-agent' (2 events over 0s) at 2020-07-21 15:48:22 +0200 CEST 
```

File: pkg/parser/enrich.go (hunk 3 above)
Comment (buixor) [reply pkg/parser/enrich.go:47]: no, the plugins are "attempted" to be loaded even if the parser itself isn't enabled.
It isn't very elegant but we have to deal with this until we transform the enrichment into proper modules/plugins ^^

**Pull request merged**